File: app/src/main/java/com/kabouzeid/gramophone/preferences/basic/EditTextPreferenceX.java
package com.kabouzeid.gramophone.preferences.basic;

import android.content.Context;
import android.util.AttributeSet;

import androidx.preference.EditTextPreference;

import com.kabouzeid.gramophone.R;


/**
 * @author Aidan Follestad (afollestad)
 */
public class EditTextPreferenceX extends EditTextPreference {

    public EditTextPreferenceX(Context context) {
        super(context);
        init(context, null);
    }

    public EditTextPreferenceX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public EditTextPreferenceX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    public EditTextPreferenceX(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        setLayoutResource(R.layout.x_preference);
    }
}

File: app/src/main/java/lib/phonograph/preference/EditTextPreferenceX.kt
/*
 * Copyright (c) 2022 chr_56 & Abou Zeid (kabouzeid) (original author)
 */
package lib.phonograph.preference

import android.content.Context
import android.util.AttributeSet
import androidx.preference.EditTextPreference
import player.phonograph.R

/**
 * @author Aidan Follestad (afollestad)
 */
class EditTextPreferenceX : EditTextPreference {
    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context,
        attrs,
        defStyleAttr,
        defStyleRes) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        layoutResource = R.layout.x_preference
    }
}